### PR TITLE
Improve --help output

### PR DESCRIPTION
**Pull request by SimonBarendse**
The help output grew pretty big and it was hard to find what you
were looking for. By only showing direct subcommands, the help
output at root (secrethub --help) becomes a lot shorter.

I've split the subcommands in management commands and commands
(similarly to the docker --help output), so that it is clear
which commands have another level of subcommands.

Flags that can be used on all commands are no longer displayed in
the help text of the specific commands.

When displaying subcommands, we only show the name and help text.
The usage can be viewed by running --help on the sub-command.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -25,6 +24,7 @@ require (
 	golang.org/x/crypto v0.0.0-20190313024323-a1f597ede03a
 	golang.org/x/sys v0.0.0-20190222072716-a9d3bda3a223
 	golang.org/x/text v0.3.0
-	gopkg.in/alecthomas/kingpin.v2 v2.2.6
 	gopkg.in/yaml.v2 v2.2.2
 )
+
+replace github.com/alecthomas/kingpin => github.com/simonbarendse/kingpin v0.0.0-20190509121241-d44931961c2f
```
Comment (florisvdg) [inline go.mod]: Nice trick.

**Comment by florisvdg** [comment]
Maybe it makes sense to sort the commands alphabetically? Or is there some logical ordering already that I'm missing?

**Comment by SimonBarendse** [comment]
Hi @florisvdg , what's your output of `secrethub --help`?

I get:
```
Management Commands:
  account  Manage your SecretHub account.
  audit    Show the audit log of all actions on a repository or a secret.
  config   Manage your local .secrethub configuration file.
  org      Manage a SecretHub organization.
  repo     Manage repositories.
  service  Manage a service account.

Commands:
  help      Show help.
  clear     Clear the secrets from your local environment. This reads and parses the secrets.yml file in the current working directory.
  inject    Read a template from stdin and write to stdout with secrets injected into the template.
  inspect   Inspect a repository or a secret.
  ls        List all contents of a path.
  mkdir     Create a new directory at a given path.
  printenv  Print debug statements about environment variables.
  read      Read a secret.
  rm        Permanently remove a directory, a secret or a secret version.
  run       Runs a program and passes environment variables to it, sourcing values from SecretHub.
  set       Set the secrets in your local environment. This reads and parses the secrets.yml file in the current working directory.
  signup    Signup for an account on SecretHub.
  tree      List contents of directories in a tree-like format.
  write     Write to a path to create or overwrite a secret. Input can be piped, copied from the clipboard or you will be asked to input the secret.


Run 'secrethub <command> --help' for more information on a command.
```

Which is alphabetically sorted, right?

**Comment by florisvdg** [comment]
Ah, the H in `help` coming before the C in `clear` confused me, never mind. 😅

**Comment by SimonBarendse** [comment]
Ahh, yes. The `help` command is added [in Kingpin](https://github.com/alecthomas/kingpin/blob/v2.2.6/app.go#L319), the other commands are added afterwards in alphabetical order in [app.go](https://github.com/secrethub/secrethub-cli/blob/v0.21.3/internals/secrethub/app.go#L125).

I think that's oke for now.

**Comment by mackenbach** [comment]
Wow, I absolutely love this change. It makes it insanely easier to find what you're looking for, especially for a new user. I guess the old help text was a result of adding features over time until it became too bloated. 

Below I have some improvements:
- I would prefer it if the commands were logically grouped on a need to know basis (e.g. start with signup, group the unix-like commands, end with more complex commands)
- `acl` command is missing
- `generate` command is missing
- `audit` command should be shown as a root command
- `write` command has an incorrect description. You cannot overwrite secrets. Only append a new version. 
- `clear` and `set commands should be hidden.
- We use long and wordy descriptions where short ones suffice. We could print longer explanations in the specific command help text. 

IMO the help text for commands should look like this:

```
Management Commands:
  org      Manage shared organization workspaces.
  repo     Manage repositories.
  acl      Manage access rules on directories
  service  Manage service accounts.
  account  Manage your personal account.
  config   Manage your local configuration file.

Commands:
  help      Show help.
  signup    Create a free personal developer account.
  write     Write a secret.
  read      Read a secret.
  generate  Generate a random secret.
  ls        List contents of a path.  
  mkdir     Create a new directory.
  rm        Remove a directory, secret or version.
  tree      List contents of a path in a tree-like format.
  inspect   Print details of a resource.
  audit     Show the audit log.  
  inject    Inject secrets into a template.
  run       Pass secrets as environment variables to a process.
  printenv  Debug environment variables.
```

Finally, the first few lines could use a tiny bit of editing to make them more readable. My proposal is to change the first lines of the help text from:

```
The SecretHub Command-Line Interface (CLI) is a unified tool to manage your
secrets inside SecretHub.

Check out https://secrethub.io/docs/getting-started/ for a step-by-step
introduction.

The CLI is configurable through command-line flags and environment variables.
Options set on the command-line take precedence over those set in the
environment. The format for environment variables is
`SECRETHUB_[COMMAND_]FLAG_NAME`.
```
To:
```
The SecretHub command-line interface is a unified tool to manage your 
infrastructure secrets with SecretHub.

For a step-by-step introduction, check out:

   https://secrethub.io/docs/getting-started/

The CLI is configurable through command-line flags and environment variables.
Options set on the command-line take precedence over those set in the
environment. The format for environment variables is
`SECRETHUB_[COMMAND_]FLAG_NAME`.
```

**Comment by SimonBarendse** [comment]
Hi @mackenbach , good suggestions! I've opened #74 to fix those.

I've changed a few things:
- `tree` lists contents of a directory
- I've not yet shown the `generate` command in the help text. Currently, the `generate` command is a management command, as `rand` is a subcomand of `generate`. Let's change that first before showing it in the help text.

**Pull request merged**